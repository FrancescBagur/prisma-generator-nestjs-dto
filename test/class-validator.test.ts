import { testFixtures } from './fixtures';

testFixtures('fixtures/class-validator.prisma');
