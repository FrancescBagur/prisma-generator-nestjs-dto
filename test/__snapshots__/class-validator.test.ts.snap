// Vitest Snapshot v1

exports[`fixtures/class-validator.prisma 1`] = `
[
  {
    "content": "
import {IsUUID} from 'class-validator'
  export class ConnectProductDto {
  @IsUUID('4')
id: string;
  }
  ",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/connect-product.dto.ts",
  },
  {
    "content": "
import {Prisma} from '@prisma/client'
import {IsUUID,IsBoolean,IsPositive,IsDecimal} from 'class-validator'


export class CreateProductDto {
  @IsUUID('4')
id: string;
  description: string;
  images: Prisma.InputJsonValue[];
  @IsBoolean()
highlighted?: boolean;
  @IsPositive()
reviewCount?: number;
  @IsDecimal({
  decimal_digits: 2
})
score?: number;
  @IsUUID('4')
categoryId?: string;
  @IsUUID('4')
companyId: string;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/create-product.dto.ts",
  },
  {
    "content": "
import {Prisma} from '@prisma/client'
import {IsUUID,IsBoolean,IsPositive,IsDecimal} from 'class-validator'


export class UpdateProductDto {
  @IsUUID('4')
id: string ;
  description?: string ;
  images?: Prisma.JsonValue[] ;
  @IsBoolean()
highlighted?: boolean  | null;
  @IsPositive()
reviewCount?: number  | null;
  @IsDecimal({
  decimal_digits: 2
})
score?: number  | null;
  @IsUUID('4')
categoryId?: string  | null;
  @IsUUID('4')
companyId?: string ;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/update-product.dto.ts",
  },
  {
    "content": "
import {Prisma} from '@prisma/client'
import {IsUUID,IsBoolean,IsPositive,IsDecimal} from 'class-validator'
import {IntersectionType} from '@nestjs/swagger'

export class Product {
  @IsUUID('4')
id: string ;
  description: string ;
  images: Prisma.JsonValue[] ;
  @IsBoolean()
highlighted: boolean  | null;
  @IsPositive()
reviewCount: number  | null;
  @IsDecimal({
  decimal_digits: 2
})
score: number  | null;
  @IsUUID('4')
categoryId: string  | null;
  @IsUUID('4')
companyId: string ;
}
export class ProductRel {
  
}
export class ProductFull extends IntersectionType(Product, ProductRel) {}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/product.entity.ts",
  },
]
`;
